app/utils/pages.tsx

```diff
@@ -13,61 +10,76 @@ interface IPage {
     path: string
 }
 
-export function getPathsToMap (){
-    const pathToLeagueData = path.join(process.cwd(), "app", "leagueConfiguration");
-    const pathsToMap = fs.readdirSync(pathToLeagueData);
-    return pathsToMap;
+interface PageInformation {
+    showStatus: string, 
+    showNameAndSeason: string,
+    friendlyName: string,
+    contestantDataKey: string
 }
 
-export function getLeagueConfigurationData (filename:string){
-    return require(`../leagueConfiguration/${filename}`);
+function constructPageInformation(leagueConfigurationKey:string){
+    const showKey = leagueConfigurationKey.replace("league_configuration:", "");
+    const params = showKey.replaceAll("_", "-").split(":");
+    const showStatus = params[0];
+    const showNameAndSeason = `${params[1]}-${params[2]}`;
+    const friendlyShowName = params[1].split("-").map(word => word.charAt(0).toUpperCase() + word.slice(1)).join(" ");
+    const friendlyName = `${friendlyShowName} ${params[2]}`;
+    const contestantDataKey = `${showKey.replace(/(archive|active):/i, "")}:*`;
+    return {
+        contestantDataKey,
+        showStatus,
+        showNameAndSeason,
+        friendlyName
+    }
+}
+
+function generateSubpages (pageInformation:PageInformation, hasContestantData:boolean){
+    const subpages:Array<IPage> = [];
+    subpages.push({
+        name: "Contestants",
+        path: `/${pageInformation.showStatus}/${pageInformation.showNameAndSeason}/contestants`
+    });
+    if(hasContestantData){
+        const scoringSubpage = {
+            name: "Scoring",
+            path: `/${pageInformation.showStatus}/${pageInformation.showNameAndSeason}/scoring`
+        }
+        const leagueStandingSubpage = {
+            name: "League Standing",
+            path: `/${pageInformation.showStatus}/${pageInformation.showNameAndSeason}/league-standing`
+        }
+        subpages.push(scoringSubpage, leagueStandingSubpage);
+    }
+    return subpages;
 }
 
-export function checkForSubpages (filename:string){
-    return fs.existsSync(path.join(process.cwd(), "app", "leagueData", filename));
+function generatePathObj(pageData:PageInformation, hasContestantData:boolean){
```

Comment:
nit: this is usually a big code smell to me. Why pass a boolean when what we really mean is we have two different functions?

Reply:
We talked about this offline and got a refactor we both were happy about